engine/kube/util.go

```diff
@@ -236,6 +238,13 @@ func toResources(step *engine.Step) v1.ResourceRequirements {
 // helper function returns a kubernetes pod for the
 // given step and specification.
 func toPod(spec *engine.Spec, step *engine.Step) *v1.Pod {
+	config, err := config.Environ()
+
+	if err != nil {
+		logrus.WithError(err).Fatalln("invalid configuration")
```

Comment:
this should return an error. `Fatalln` will cause kill the current running program.

Reply:
This has been removed in refactoring to use step config!